### DISCUSSION TITLE
Is there a way to view input relations?

### DISCUSSION THREAD

**Discussion by doug-galvanick** [vmware/differential-datalog#1166] [Q&A] [answered]
In the subnet example in the tutorial, the output relation has two UUIDs. Is there a way to back reference those UUIDs to get the input relations they're associated with? Or would that be handled by an external program? Looking for something like `dump Host where ...`.

Thanks!

```
// Type aliases
typedef UUID    = bit<128>
typedef IP4     = bit<32>
typedef NetMask = bit<32>

//IP host specified by its name and address.
input relation Host(id: UUID, name: string, ip: IP4)

//IP subnet specified by its IP prefix and mask
input relation Subnet(id: UUID, prefix: IP4, mask: NetMask)

// HostInSubnet relation maps hosts to known subnets
output relation HostInSubnet(host: UUID, subnet: UUID)

HostInSubnet(host_id, subnet_id) :- Host(host_id, _, host_ip),
                                    Subnet(subnet_id, subnet_prefix, subnet_mask),
                                    ((host_ip & subnet_mask) == subnet_prefix). // filter condition
```

**Comment by doug-galvanick** [comment, marked as answer]
It looks like the answer to this is to use indexes in conjunction with the `query_index` command.

https://github.com/vmware/differential-datalog/blob/master/doc/tutorial/tutorial.md#indexes

https://github.com/vmware/differential-datalog/blob/master/doc/command_reference/command_reference.md

**Reply by ryzhyk** [reply]
That's right. This is not as general as querying with an arbitrary `where` clause, but is the closest thing we have to that.

**Reply by doug-galvanick** [reply]
Thanks!
